File: python/2965.Find-Missing-and-Repeated-Values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        total = n * n
        freq = [0] * (total + 1)
        missing = None
        repeating = None

        for r in range(n):
            for c in range(n):
                freq[grid[r][c]] += 1

        for i in range(1, total + 1):
            if freq[i] == 0:
                missing = i
            if freq[i] > 1:
                repeating = i

        return [repeating, missing]

File: javascript/2965.Find-Missing-and-Repeated-Values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function (grid) {
  const n = grid.length;
  const total = n * n;
  const freq = new Array(total + 1).fill(0);
  let repeating;
  let missing;

  for (let r = 0; r < n; r++) {
    for (let c = 0; c < n; c++) {
      freq[grid[r][c]] += 1;
    }
  }

  for (let i = 1; i <= total; i++) {
    if (freq[i] === 0) {
      missing = i;
    }

    if (freq[i] > 1) {
      repeating = i;
    }
  }

  return [repeating, missing];
};
